pytest_mozwebqa/driver.py

```diff
@@ -0,0 +1,174 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+
+import pytest
+from selenium.webdriver.support.event_firing_webdriver import \
+    EventFiringWebDriver
+from selenium.webdriver.common.proxy import Proxy
+from selenium.webdriver.firefox.firefox_binary import FirefoxBinary
+from selenium import webdriver
+
+
+def proxy_from_options(options):
+    if options.proxy_host and options.proxy_port:
+        proxy_str = '%(proxy_host)s:%(proxy_port)s' % vars(options)
+        proxy = Proxy()
+        proxy.ssl_proxy = proxy.http_proxy = proxy_str
+        return proxy
+
+
+def make_driver(item):
+    options = item.config.option
+    capabilities = {}
+    if options.capabilities is not None:
+        for c in options.capabilities:
+            name, value = c.split(':')
+            # handle integer capabilities
+            if value.isdigit():
+                value = int(value)
+            # handle boolean capabilities
+            elif value.lower() in ['true', 'false']:
+                value = value.lower() == 'true'
+            capabilities[name] = value
+    proxy = proxy_from_options(options)
+    if proxy is not None:
+        proxy.add_to_capabilities(capabilities)
+    return start_driver(item, options, capabilities)
+
+
+def start_driver(item, options, capabilities):
+    specific_driver = '%s_driver' % options.driver.lower()
+    make_driver = globals().get(specific_driver, generic_driver)
```

Comment:
Sure, do you have a suggestion? Much of this code was from Ronny's original patch, but I would agree that it can be a little hard to read.